src/app_service/service/saved_address/service.nim

```diff
@@ -0,0 +1,49 @@
+import chronicles, sequtils, json
+
+import ./service_interface, ./dto
+
+import status/statusgo_backend_new/saved_addresses as backend
+
+export service_interface
+
+logScope:
+  topics = "saved-address-service"
+
+type 
+  Service* = ref object of service_interface.ServiceInterface
+    savedAddresses: seq[SavedAddressDto]
+
+method delete*(self: Service) =
+  discard
+
+proc newService*(): Service =
+  result = Service()
+
+method init*(self: Service) =
+  try:
+    let response = backend.getSavedAddresses()
+
+    self.savedAddresses = map(
+      response.result.getElems(),
+      proc(x: JsonNode): SavedAddressDto = toSavedAddressDto(x)
+    )
+
+  except Exception as e:
+    error "error: ", methodName="init", errName = e.name, errDesription = e.msg
+
+method getSavedAddresses(self: Service): seq[SavedAddressDto] =
+  return self.savedAddresses
+
+method addSavedAddress(self: Service, name, address: string) =
+  try:
+    backend.addSavedAddress(name, address)
```

Comment:
You need to add the address to `savedAddresses`, refetch or use a `isDirty` mechanism like a cache

Reply:
I will do that when I plug it to QML